picocom.c

```diff
@@ -1616,6 +1616,10 @@ main(int argc, char *argv[])
 	init_history();
 #endif
 
+	r = term_get_baudrate(tty_fd, NULL);
+	if ( r != opts.baud )
+		fd_printf(STO, "WARNING: Desired baudrate is %d but applied baudrate is %d; this might depend upon your OS and/or hardware limitations.\r\n\r\n", opts.baud, r);
+
```

Comment:
Don't like this either: If we go down this way then we have to warn the user for ALL settings that do not match due to hardware limitations (why is baudrate special?). The user can always issue a "see options command".